src/commands/create/contract.ts

```diff
@@ -458,28 +448,41 @@ export default class Contract extends Command {
+              ['string', 'string', 'uint16', 'uint256', 'bool', 'bytes'],
               [
-                'tuple(address,address,address,string,string,address,address,uint64,uint16,bytes[],address,bytes)',
-                'bool',
-              ],
-              [
-                [
-                  '0x0000000000000000000000000000000000000000', // TODO: holographFeeManager - this.networkMonitor.holographFeeManager.address
-                  '0x0000000000000000000000000000000000000000', // holographERC721TransferHelper
-                  '0x000000000000AAeB6D7670E522A718067333cd4E', // marketFilterAddress
-                  collectionName, // contractName
-                  collectionSymbol, // contractSymbol
-                  userWallet.address, // initialOwner
-                  userWallet.address, // fundsRecipient
-                  numOfEditions, // number of editions
-                  royaltyBps, // royalty percentage in bps
-                  setupCalls, // setupCalls (used to set sales config)
-                  metadataRenderer.address, // metadata renderer address deployed from above
-                  generateInitCode(['string', 'string', 'string'], [description, imageURI, animationURI]), // metadataRendererInit
-                ],
-                false, // skipInit
+                collectionName, // string memory contractName
+                collectionSymbol, // string memory contractSymbol
+                royaltyBps, // uint16 contractBps
+                allEventsEnabled(), // uint256 eventConfig -  all 32 bytes of f
+                false, // bool skipInit
+                generateInitCode(
+                  ['bytes32', 'address', 'bytes'],
+                  [
+                    // eslint-disable-next-line unicorn/prefer-string-slice
+                    '0x' + web3.utils.asciiToHex('HolographDropsEditionsV1').substring(2).padStart(64, '0'),
+                    this.networkMonitor.registryAddress,
+                    generateInitCode(
+                      [
+                        'tuple(address,address,address,address,uint64,uint16,tuple(uint104,uint32,uint64,uint64,uint64,uint64,bytes32),address,bytes)',
+                      ],
+                      [
+                        [
+                          '0x0000000000000000000000000000000000000000', // erc721TransferHelper
+                          '0x000000000000AAeB6D7670E522A718067333cd4E', // marketFilterAddress (opensea)
+                          userWallet.address, // initialOwner
+                          userWallet.address, // fundsRecipient
+                          numOfEditions, // number of editions
+                          royaltyBps, // percentage of royalties in bps
+                          salesConfig,
+                          metadataRenderer.address, // metadataRenderer
+                          generateInitCode(['string', 'string', 'string'], [description, imageURI, animationURI]), // metadataRendererInit
```

Comment:
Can we start making functions for things that generate byte code? For example

```js

function genrateFullDropBytes({ description, imageURI, etc...}) {

  const metadataRendererInitBytes = genMetaDataRenderInitBytes({description, imageURI, animationURI}) // LINE:  478
  const contractsBytes = genContractBytes({ initOwner, fundsRecipent, metadataRendererInitBytes}) // LINE: 466 inner tuple
  const dropTuple = genMoreBytes( { ..., contractsBytes, etc... }) // LINE: 464
  const otherTuple = genDropBytes({ dropTuple, ...}) // LINE: 458
  // etc...
 
  return fullBytes
}
```

Basically, can we have smaller functions that we can compose to do all of this work. The code above was only illustrate, i am open to name changes or structure.

Reply:
@ACC01ADE and I were actually talking about exactly this. I can create a helper function for composing this stuff, but it isn't highly abstract-able since the bytecode format is so specific per contract. I'll experiment a bit